pkg/api/validation/validation.go

```diff
@@ -1603,6 +1603,21 @@ func ValidateSecret(secret *api.Secret) errs.ValidationErrorList {
 		if err := json.Unmarshal(dockercfgBytes, &map[string]interface{}{}); err != nil {
 			allErrs = append(allErrs, errs.NewFieldInvalid(fmt.Sprintf("data[%s]", api.DockerConfigKey), "<secret contents redacted>", err.Error()))
 		}
+	case api.SecretTypeBasicAuth:
```

Comment:
If Kubernetes itself is not consuming these secret types in any way, then we shouldn't validate them in the apiserver.